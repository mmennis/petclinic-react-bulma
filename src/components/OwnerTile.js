import React from 'react'
import { Tile, Heading, Content, Card } from 'react-bulma-components'
import PropTypes from 'prop-types'

export default class OwnerTile extends React.Component {

    render() {
        return (
            <Tile size={3} renderAs="article" kind="child" notification color="light" paddingless={true} key={this.props.owner._id}>
                <Card paddingless={true} rounded="true" outlined="true" >
                    <Card.Header outlined="true" >
                        <Card.Header.Title>{this.props.owner.first_name} {this.props.owner.last_name}</Card.Header.Title>
                    </Card.Header>
                    <Card.Content paddingless={false} color="light">            
                        <Heading size={6} subtitle>{this.props.owner.pets.length} pets</Heading>
                        <Content >
                            <Heading marginless={true} paddingless={false} size={6}>Address:</Heading>
                            <p>{this.props.owner.address}<br/>{this.props.owner.city}, {this.props.owner.state}</p>
                        </Content>
                        <Content size="small">
                            <Heading marginless={true} paddingless={false} size={6}>Phone:</Heading >
                            <p>{this.props.owner.telephone}</p>
                        </Content>
                    </Card.Content>
                </Card>                
            </Tile>
        )
    }

}

OwnerTile.propTypes = {
    owner: PropTypes.object.isRequired
}