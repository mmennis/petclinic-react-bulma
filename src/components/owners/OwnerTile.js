import React from 'react'
import { Tile, Heading, Content, Card, Button } from 'react-bulma-components'
import PropTypes from 'prop-types'
import OwnersDetail from './OwnersDetail'
import EditOwnerForm from './EditOwnerForm' 
import Pluralize from 'react-pluralize'
import axios from 'axios'
import { data } from './owers.localisation.data'
import LocalizedStrings from 'react-localization'
import { LanguageContext } from '../localization/LanguageContext'

const strings = new LocalizedStrings(data)

const BASE_URL = process.env.REACT_APP_PETCLINIC_APP_API_ENDPOINT;

export default class OwnerTile extends React.Component {

    constructor(props) {
        super(props)

        this.state ={
            updated: false
        }
        this.handleOwnerUpdate = this.handleOwnerUpdate.bind(this)
        this.handleDelete = this.handleDelete.bind(this)
    }

    handleOwnerUpdate = (owner) => {
        console.log(`Received an updated owner: ${JSON.stringify(owner)}`)
        axios.put(`${BASE_URL}/owners/${owner._id}`, owner)
            .then((response) => {
                console.log(`Response from owner update: ${JSON.stringify(response)}`)
            })
            .catch((err) => {
                console.error(`Problem updating owner: ${err}`)
            })
            .finally(() => {
                console.log(`Owner ${owner._id} updated`)
            })
    }

    handleDelete = () => {
        this.props.handleDeleteOwner(this.props.owner)
    }

    render() {
        const localeCtx = this.context
        strings.setLanguage(localeCtx.lang)
        return (
            <Tile size={3} renderAs="article" kind="child" notification color="light" style={{ 'padding': '10px 20px 10px 20px'}}paddingless={false} key={this.props.owner._id}>
                <Card paddingless={true} rounded="true" outlined="true" >
                    <Card.Header outlined="true" >
                        <Card.Header.Title>{this.props.owner.first_name} {this.props.owner.last_name}</Card.Header.Title>
                    </Card.Header>
                    <Card.Content paddingless={false} color="light">            
                        <Content size="small">
                            <p>{this.props.owner.city}, {this.props.owner.state}</p>
                        </Content>
                        <Heading size={4} subtitle><Pluralize singular={'pet'} count={this.props.owner.pets.length}/></Heading>
                    </Card.Content>
                    <Card.Footer>
                        <OwnersDetail owner={this.props.owner} modal={{closeOnBlur: true, showClose: true }}/>
                        <EditOwnerForm 
                            owner={this.props.owner} 
                            handleOwnerUpdate={this.handleOwnerUpdate} 
                            modal={{closeOnBlur: true, showClose: true }}
                        />
                        <Button 
                            onClick={this.handleDelete}
                            color="danger" size="small"
                            data-testid="delete-button"
                            rounded={true}
                        >
                            {strings.delete_button}
                        </Button>
                    </Card.Footer>
                </Card>                
            </Tile>
        )
    }

}

OwnerTile.contextType = LanguageContext

OwnerTile.propTypes = {
    owner: PropTypes.object.isRequired,
    handleDeleteOwner: PropTypes.func.isRequired,
}