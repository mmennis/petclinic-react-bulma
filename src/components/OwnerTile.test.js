import React from 'react'
import { create } from 'react-test-renderer'
import { render } from '@testing-library/react'
import OwnerTile from './OwnerTile'

const testOwner = {
    first_name: 'First',
    last_name: 'Last',
    address: '1234 Main St.',
    city: 'Cupertino',
    state: 'CA',
    telephone: '408-555-1212',
    pets: ['fido']
}

describe('OwnerTile', () => {
    it('should render as per the snapshot', () => {
        const ownerTile = create(<OwnerTile owner={testOwner} />)
        expect(ownerTile.toJSON()).toMatchSnapshot()
    })

    it('should have owner details in the output', () => {
        console.error = jest.fn()
        const { getByText } = render(<OwnerTile owner={testOwner} />)
        const ownerFName = getByText(/First/i)
        expect(ownerFName).toBeInTheDocument()
        const ownerLName = getByText(/Last/i)
        expect(ownerLName).toBeInTheDocument()
        const city = getByText(/Cupertino/i)
        expect(city).toBeInTheDocument()
    })

})