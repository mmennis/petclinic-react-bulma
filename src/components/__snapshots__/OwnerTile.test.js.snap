// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerTile should render as per the snapshot 1`] = `
<article
  className="tile notification is-child is-3 is-light is-paddingless"
>
  <div
    className="card is-paddingless"
    outlined={true}
  >
    <div
      className="card-header"
      outlined={true}
    >
      <div
        className="card-header-title"
      >
        First
         
        Last
      </div>
    </div>
    <div
      className="card-content"
      color="light"
    >
      <h1
        className="subtitle is-6"
      >
        1
         pets
      </h1>
      <div
        className="content"
      >
        <h1
          className="title is-6 is-marginless"
        >
          Address:
        </h1>
        <p>
          1234 Main St.
          <br />
          Cupertino
          , 
          CA
        </p>
      </div>
      <div
        className="content is-small"
      >
        <h1
          className="title is-6 is-marginless"
        >
          Phone:
        </h1>
        <p>
          408-555-1212
        </p>
      </div>
    </div>
  </div>
</article>
`;
